url1cbaffb197_foxking0416_AlgorithmTraining_tgz-pJ8-algorithm_leetcode_MainLeetJ8/src/datastructure/binarytree/BinaryTree.java

```diff
@@ -0,0 +1,1106 @@
+			return false;
+		
+		result &= (p.val == q.val);
+		if(result == false)
+			return false;
+		
+		
+		 if((p.getLeftLeafNode() == null && q.getLeftLeafNode() != null) || (p.getLeftLeafNode() != null && q.getLeftLeafNode() == null))
+			return false;
+		 else if((p.getLeftLeafNode() != null && q.getLeftLeafNode() != null) && (!isSameTree(p.getLeftLeafNode(), q.getLeftLeafNode())))
+			 return false;
+		
+		 if((p.getRightLeafNode() == null && q.getRightLeafNode() != null) || (p.getRightLeafNode() != null && q.getRightLeafNode() == null))
+			return false;
+		 else if((p.getRightLeafNode() != null && q.getRightLeafNode() != null) && (!isSameTree(p.getRightLeafNode(), q.getRightLeafNode())))
+			 return false;
+		
+
+		
+		return true;
+    }
+
+    public boolean hasPathSum(TreeNode root, int sum) {
+        
+    	if(root.getLeftLeafNode() == null && root.getRightLeafNode() == null){
+    		if(sum == root.val)
+    			return true;
+    		else 
+				return false;
+    	}
+    	
+    	boolean resultLeft = false;
+    	boolean resultRight = false;
+    	
+    	if(root.getLeftLeafNode() != null){
+    		resultLeft = hasPathSum(root.getLeftLeafNode(), sum - root.val);
+    	}
+    	if(root.getRightLeafNode() != null){
+    		resultRight = hasPathSum(root.getRightLeafNode(), sum - root.val);
+    	}
+    	
+    	return (resultLeft | resultRight);
+    }
+    
+    public List<List<Integer>> pathSumII(TreeNode root, int sum) {
+        List<List<Integer>> result = new ArrayList<List<Integer>>();
+        List<Integer> path = new ArrayList<Integer>();
+        
+        if(root == null)
+        	return result;
+
+        pathSumHelper(root, sum, path, result);
+        
+        return result;
+    }
+    
+    private void pathSumHelper(TreeNode root, int sum, List<Integer> path, List<List<Integer>> totalPathResult) {
+    
+    	//if(root != null)
+    		path.add(root.val);
+		
+		if(root.getLeftLeafNode() == null && root.getRightLeafNode() == null){
+			if(sum == root.val){
+				List<Integer> pathToStore = new ArrayList<Integer>(path);
+				totalPathResult.add(pathToStore);
+			}
+	    	path.remove(path.size()-1);
+			return;
+		}
+
+    
+    	
+    	if(root.getLeftLeafNode() != null){
+    		pathSumHelper(root.getLeftLeafNode(), sum - root.val, path, totalPathResult);
+    	}
+    	if(root.getRightLeafNode() != null){
+    		pathSumHelper(root.getRightLeafNode(), sum - root.val,  path, totalPathResult);
+    	}
+    	
+    	path.remove(path.size()-1);
+	}
+    
+    public static boolean isSymmetric(TreeNode root) {
+        
+    	if(root == null)
+    		return true;
+    	
+    	if(root.getLeftLeafNode() != null && root.getRightLeafNode() == null){
+    		return false;
+    	}
+    	if(root.getLeftLeafNode() == null && root.getRightLeafNode() != null){
+    		return false;
+    	}
+    	
+    	return symmetric(root.getLeftLeafNode(), root.getRightLeafNode());
+    }
+    
+    private static boolean symmetric(TreeNode p, TreeNode q){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 706-713 into a separate method.